type-annotator-scanner/README.md

```diff
@@ -1 +1,100 @@
-## Type Annotator Scanner (Scanner)
\ No newline at end of file
+## Type Annotator Scanner (Scanner)
+
+`Type Annotator Scanner` is an Error Prone checker which serializes the required information regarding the source code structure for the Annotator.
+
+Below are the series of information serialized by this checker:
+1. `Region` info where an element in the source code is used.
+2. Method info to construct the `Method Inheritance Tree`.
+3. Class info.
+
+See sections below where each series is described in details
+
+#### Region
+
+An __element__ in the source code is __used__ if:
+- __element__ is a method and the method is invoked.
+- __element__ is a class field and the field is read/initialized.
+
+
+A __Region__ is the enclosing method body where an element is used. Each region is denoted as below:
+1. `method`: Enclosing method signature in `string`.
+2. `class`: Enclosing class flat name in `string`.
+
+If the program point where the element is used is not enclosed by a method (e.g. class initialization region) the `method` value be the string `"null"`.
+
+#### Method Inheritance Tree
+`Method Inheritance Tree` is a tree of nodes where each node corresponds to a method in the source node. Each node's parent in this structure, is the node which corresponds to method's closest overriding method.
+
+A node corresponding to method `m` contains the information below:
+1. Signature of `m`.
+2. Flat name of the enclosing class of `m`.
+3. Annotations of each parameter.
+4. Annotation of the method return.
+5. Unique id of this node.
+6. Unique id of parent node.
+7. Visibility of `m`.
+8. Return type of `m`.
+
+#### Class info
+Serializes flat name of all observed classes and uri of the file containing the class.
+
+These data is used by the `core` module to detect multiple inline field declarations.
```

Comment:
"This data" (uncountable noun)

Reply:
1e2286bae6c6cc4f36c86d81d760df0f80c294fd